proxy_http.go

```diff
@@ -116,19 +116,22 @@ func (proxy *proxy) handle(ctx context.Context, downstreamIn io.Reader, downstre
 	if req.Method == http.MethodConnect {
 		next = proxy.nextCONNECT(downstream)
 	} else {
-		var tr *http.Transport
+		var tr idleClosingTransport
```

Comment:
Do you need to get this address-logging wrapper rigged into the case where there is not an established upstream connection (upstream == nil) as well or does that pass back through here somehow ?

Reply:
I can't think of a way to rig it in unless I fork net/http.

Reply:
fair enough. no harm if this part is useful I suppose -- when would the other case be triggered?

Reply:
Sadly, it's the usual case. The case with the existing connection only happens when we're MITM'ing a connection.